modules/gateway/peers.go

```diff
@@ -251,29 +303,11 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 	if err != nil {
 		return err
 	}
-	// send our version
-	if err := encoding.WriteObject(conn, build.Version); err != nil {
-		return err
-	}
-	// read version ack
-	var remoteVersion string
-	if err := encoding.ReadObject(conn, &remoteVersion, maxAddrLength); err != nil {
-		return err
-	}
-	// decide whether to accept this version
-	if remoteVersion == "reject" {
-		return errPeerRejectedConn
-	}
-	// Check that version is acceptable.
-	//
-	// Reject peers < v0.4.0 as the previous version is v0.3.3 which is
-	// pre-hardfork.
-	//
-	// NOTE: this version must be bumped whenever the gateway or consensus
-	// breaks compatibility.
-	if build.VersionCmp(remoteVersion, "0.4.0") < 0 {
+	remoteVersion, err := connectVersionHandshake(conn, build.Version)
+	if err != nil {
+		// TODO: should this be muxado.Client(conn).Close()?
 		conn.Close()
```

Comment:
The error returned by `session.Accept()` on `Close` is the same whether the connection is closed with `conn.Close()` or `muxado.Client(conn).Close()`. Since we'll be switching to YAMUX / SPDY / HTTP2 soon anyways it is probably easier to continue closing with `conn.Close()`.